### PR TITLE
Placeholder alignment options, clearButtonMode, option for showing/hiding of the password toggle and minor bug fixes.

### PR DESCRIPTION
**Pull request by sasojadrovski**
Added an option for setting the placeholder alignment.
Fixed an issue where the caret is improperly aligned the first time
`becomeFirstResponder` is called when the `textAlignment` property is
changed.

### CODE REVIEW

**Comment by jorgerebolloj** [comment]
@sasojadrovski Greetings, how can I configure the AnimatedTextInput for alignment to the center with the implementations that you implemented? Thanks in advance

File: AnimatedTextInput/Classes/AnimatedTextInput.swift [hunk 1]
```diff
@@ -157,8 +169,14 @@ open class AnimatedTextInput: UIControl {
     fileprivate var placeholderErrorText: String?
 
     fileprivate var placeholderPosition: CGPoint {
-        let hintPosition = CGPoint(x: style.leftMargin, y: style.yHintPositionOffset)
-        let defaultPosition = CGPoint(x: style.leftMargin, y: style.topMargin + style.yPlaceholderPositionOffset)
+        let hintPosition = CGPoint(
+            x: placeholderAlignment != .natural ? 0 : style.leftMargin,
```
Comment (victorBaro) [inline AnimatedTextInput/Classes/AnimatedTextInput.swift:173]: I'd rather see "positive" condition statements: 
`placeholderAlignment == .natural ? style.leftMargin : 0`

File: AnimatedTextInput/Classes/AnimatedTextInput.swift [hunk 2]
```diff
@@ -157,8 +169,14 @@ open class AnimatedTextInput: UIControl {
...
+            y: style.yHintPositionOffset
+        )
+        let defaultPosition = CGPoint(
+            x: placeholderAlignment != .natural ? 0 : style.leftMargin,
```
Comment (victorBaro) [inline AnimatedTextInput/Classes/AnimatedTextInput.swift:177]: Same

File: AnimatedTextInput/Classes/AnimatedTextInputFieldConfigurator.swift [hunk 3]
```diff
@@ -5,7 +5,7 @@ public struct AnimatedTextInputFieldConfigurator {
     public enum AnimatedTextInputType {
         case standard
         case email
-        case password
+        case password(toggleable: Bool)
```
Comment (victorBaro) [inline AnimatedTextInput/Classes/AnimatedTextInputFieldConfigurator.swift:8]: I like that 😍

**Review by victorBaro — APPROVED** [review]
Love the new additions :clap:
I only had a minor code style comment, but merging anyway. Pretty excited about all the new stuff. 
Many thanks for contributing.

File: AnimatedTextInput/Classes/AnimatedTextInput.swift (hunk 1 above)
Comment (sasojadrovski) [reply AnimatedTextInput/Classes/AnimatedTextInput.swift:173]: I will take it into consideration in the future :) I agree with you!

File: AnimatedTextInput/Classes/AnimatedTextInputFieldConfigurator.swift (hunk 3 above)
Comment (sasojadrovski) [reply AnimatedTextInput/Classes/AnimatedTextInputFieldConfigurator.swift:8]: Happy to hear that! 😄

**Comment by sasojadrovski** [comment]
@victorBaro 
Thank you for the awesome library! I really like the way it is developed and structured. It is easily understandable and therefore easy to contribute to. I am more than happy that you liked and merged my pull request. It means a lot! 

Keep up the great work, and I will continue contributing in case I have more ideas about new features and/or improvements :)

**Pull request merged**